test/manipulation.test.js

```diff
@@ -875,6 +877,66 @@ describe('manipulation', function () {
         e.should.be.eql(new Error('Invalid date: X'));
       }
     });
+  });
+
+  describe('update/updateAll', function() {
+    var connectors = ['memory', 'transient'];
+
+    function getModel(connector) {
+      var ds = getSchema(connector);
+      return ds.define('Model');
+    }
 
+    it('should implement the `updateAll` function', function() {
+      connectors.forEach(function(connector) {
+        var Model = getModel(connector);
+        Model.update.should.be.a.Function;
+        Model.updateAll.should.be.a.Function;
+      });
+    });
+
+    it('should not update the model when no match is found', function(done) {
+      async.forEach(connectors, function(connector, callback) {
+        var Model = getModel(connector);
+        async.waterfall([
+          function setup(cb) {
+            Model.create({id: 1, name: 'foo'}, cb);
+          },
+          function verify(model, cb) {
+            Model.update({id: 1, name: 'bar'}, {name: 'baz'}, function(err,
+                count) {
+              should.not.exist(err);
+              model.name.should.equal('foo');
+              count.should.be.a.Number.and.be.exactly(0);
+              cb();
+            });
+          }
+        ], callback);
+      }, done);
+    });
+
+    it.only('should update the model when a match is found', function(done) {
+      // skip the transient connector
+      var index = connectors.indexOf('transient');
+      connectors.splice(index, 1);
+
+      async.forEach(connectors, function(connector, callback) {
+        var Model = getModel(connector);
+        async.waterfall([
+          function setup(cb) {
+            Model.create({id: 1, name: 'foo'}, cb);
+          },
+          function verify(model, cb) {
+            model.name.should.equal('foo');
+            Model.update({name: 'foo'}, {name: 'baz'}, function(err, count) {
+              model.name.should.equal('baz');
```

Comment:
@bajtos This test is failing. I'm not sure why yet, but it doesn't rename `model`.